### PR TITLE
Feature/new collections view

### PR DESCRIPTION
**Pull request by OlegGordiichuk**
- Add new ui for collectables
- Refactor code 
- Code cleaning
- Code optimization

### CODE REVIEW

File: Trust/Tokens/ViewModels/NonFungibleCollectionViewCellModel.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    var name: String {
+        return token.name
+    }
+    
+    var annotation: String {
+        return token.annotation
+    }
+
+    var imagePath: URL? {
+        return URL(string: token.imagePath)
```
Comment (vikmeup) [inline Trust/Tokens/ViewModels/NonFungibleCollectionViewCellModel.swift]: can you use `token.imageURL`?

File: Trust/Tokens/ViewControllers/NonFungibleTokensViewController.swift [hunk 2]
```diff
@@ -97,10 +71,11 @@ class NonFungibleTokensViewController: UIViewController {
 
     func fetch() {
         startLoading()
-        viewModel.fetchAssets { state in
+        viewModel.fetchAssets { [weak self] state in
             if state {
-                self.endLoading()
+                self?.endLoading()
```
Comment (vikmeup) [inline Trust/Tokens/ViewControllers/NonFungibleTokensViewController.swift]: should we call `endLoading(error:)` is state is false?

File: Trust/Tokens/Coordinators/TokensCoordinator.swift [hunk 3]
```diff
@@ -60,6 +60,15 @@ class TokensCoordinator: Coordinator {
         self.store = tokensStorage
         self.network = network
         self.transactionsStore = transactionsStore
+
+        NotificationCenter.default.addObserver(self, selector: #selector(self.showSpinningWheel(_:)), name: NSNotification.Name(rawValue: "ShowToken"), object: nil)
```
Comment (vikmeup) [inline Trust/Tokens/Coordinators/TokensCoordinator.swift]: What is this used for?

File: Trust/Tokens/Views/NonFungibleTokenViewCell.swift [hunk 4]
```diff
@@ -5,16 +5,45 @@ import Kingfisher
 
 class NonFungibleTokenViewCell: UITableViewCell {
 
-    @IBOutlet weak var tokenImageView: UIImageView!
-    @IBOutlet weak var annotationTextLabel: UILabel!
-    @IBOutlet weak var nameTextLabel: UILabel!
+    @IBOutlet private weak var collectionView: UICollectionView!
+
+    fileprivate var viewModel: NonFungibleTokenCellViewModel?
```
Comment (vikmeup) [inline Trust/Tokens/Views/NonFungibleTokenViewCell.swift]: can you make viewModel non-optional?

File: Trust/Tokens/Views/NonFungibleTokenViewCell.swift [hunk 5]
```diff
@@ -5,16 +5,45 @@ import Kingfisher
...
+
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        collectionView.dataSource = self
+        collectionView.delegate = self
+        collectionView.register(R.nib.nonFungibleCollectionViewCell(), forCellWithReuseIdentifier: R.nib.nonFungibleCollectionViewCell.name)
+        collectionView.backgroundColor = UIColor.clear
```
Comment (vikmeup) [inline Trust/Tokens/Views/NonFungibleTokenViewCell.swift]: would it be easier to specify clear color from the xib itself?

File: Trust/Tokens/Views/NonFungibleCollectionViewCell.swift [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        super.awakeFromNib()
+        self.layer.shadowColor = UIColor.black.cgColor
+        self.layer.shadowOffset = CGSize(width: 0, height: 1.0)
+        self.layer.shadowRadius = 2
+        self.layer.shadowOpacity = 0.5
+        self.layer.masksToBounds = false
+        imageViewBackground.backgroundColor = UIColor.random()
+    }
+
+    func config(with viewModel: NonFungibleCollectionViewCellModel) {
```
Comment (vikmeup) [inline Trust/Tokens/Views/NonFungibleCollectionViewCell.swift]: can we call configure() for consistency?

File: Trust/Tokens/ViewModels/NonFungibleCollectionViewCellModel.swift (hunk 1 above)
Comment (OlegGordiichuk) [reply Trust/Tokens/ViewModels/NonFungibleCollectionViewCellModel.swift]: Done

File: Trust/Tokens/ViewControllers/NonFungibleTokensViewController.swift (hunk 2 above)
Comment (OlegGordiichuk) [reply Trust/Tokens/ViewControllers/NonFungibleTokensViewController.swift]: Done

File: Trust/Tokens/Coordinators/TokensCoordinator.swift (hunk 3 above)
Comment (OlegGordiichuk) [reply Trust/Tokens/Coordinators/TokensCoordinator.swift]: We need this to be able to navigate to the collectable info. Yeah we should refactor this.

File: Trust/Tokens/Views/NonFungibleTokenViewCell.swift (hunk 4 above)
Comment (OlegGordiichuk) [reply Trust/Tokens/Views/NonFungibleTokenViewCell.swift]: For now it is quite fine I will think how can I refactor this.

File: Trust/Tokens/Coordinators/TokensCoordinator.swift (hunk 3 above)
Comment (vikmeup) [reply Trust/Tokens/Coordinators/TokensCoordinator.swift]: This looks very hacky. can you create an issue to refactor it?

Comment (OlegGordiichuk) [reply Trust/Tokens/Coordinators/TokensCoordinator.swift]: Done

**Pull request merged**